parquet-avro/src/test/java/org/apache/parquet/avro/TestReadVariant.java

```diff
@@ -668,8 +686,8 @@ public void testValueAndTypedValueConflict() throws IOException {
 
     assertThrows(
         () -> writeAndRead(schema, record),
-        IllegalArgumentException.class,
-        "Invalid variant, conflicting value and typed_value");
+        IllegalStateException.class,
```

Comment:
There wasn't a good place to catch and re-throw a better exception, so I used the builder's exception directly here.